platforms/jvm/toolchains-jvm-shared/src/main/java/org/gradle/jvm/toolchain/JvmImplementation.java

```diff
@@ -22,26 +22,26 @@
  *
  * @since 6.8
  */
-public final class JvmImplementation {
+public enum JvmImplementation {
```

Comment:
@octylFractal Thoughts here?

Reply:
This should be fine, I was going the other way around which is much more problematic.

Personally I feel it would be better to make this a class-enum pair similar to how `CopyOption`/`StandardCopyOption` work in the JDK. That way, it can be extended for plugins later if they add new implementations.